applications/IgaApplication/custom_conditions/sbm_dirichlet_laplacian_condition.cpp

```diff
@@ -0,0 +1,378 @@
+
+        if (mDim == 2) {
+            for (IndexType n = 1; n <= mBasisFunctionsOrder; n++) {
+                // Retrieve the appropriate derivative for the term
+                Matrix& r_shape_function_derivatives = shape_function_derivatives[n-1];
+                for (IndexType k = 0; k <= n; k++) {
+                    IndexType n_k = n - k;
+                    double derivative = r_shape_function_derivatives(i,k); 
+                    // Compute the Taylor term for this derivative
+                    H_taylor_term += ComputeTaylorTerm(derivative, mDistanceVector[0], n_k, mDistanceVector[1], k);
+                }
+            }
+        } else {
+            // 3D
+            for (IndexType n = 1; n <= mBasisFunctionsOrder; n++) {
+                Matrix& r_shape_function_derivatives = shape_function_derivatives[n-1];
+                
+                int countDerivativeId = 0;
+                // Loop over blocks of derivatives in x
+                for (IndexType k_x = n; k_x >= 0; k_x--) {
+                    // Loop over the possible derivatives in y
+                    for (IndexType k_y = n - k_x; k_y >= 0; k_y--) {
+                        
+                        // derivatives in z
+                        IndexType k_z = n - k_x - k_y;
+                        double derivative = r_shape_function_derivatives(i,countDerivativeId); 
+
+                        H_taylor_term += ComputeTaylorTerm3D(derivative, mDistanceVector[0], k_x, mDistanceVector[1], k_y, mDistanceVector[2], k_z);
+                        countDerivativeId++;
+                    }
+                }
+            }
+        }
+        H_sum_vec(i) = H_taylor_term + N(0,i);
+    }
+}
+
+unsigned long long SbmDirichletLaplacianCondition::factorial(IndexType n) 
+{
+    if (n == 0) return 1;
+    unsigned long long result = 1;
+    for (IndexType i = 2; i <= n; ++i) result *= i;
+    return result;
+}
+
+// Function to compute a single term in the Taylor expansion
+double SbmDirichletLaplacianCondition::ComputeTaylorTerm(double derivative, double dx, IndexType n_k, double dy, IndexType k)
+{
+    return derivative * std::pow(dx, n_k) * std::pow(dy, k) / (factorial(k) * factorial(n_k));    
+}
+
+double SbmDirichletLaplacianCondition::ComputeTaylorTerm3D(double derivative, double dx, IndexType k_x, double dy, IndexType k_y, double dz, IndexType k_z)
+{   
+    return derivative * std::pow(dx, k_x) * std::pow(dy, k_y) * std::pow(dz, k_z) / (factorial(k_x) * factorial(k_y) * factorial(k_z));    
+}
+
+
+int SbmDirichletLaplacianCondition::Check(const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_ERROR_IF_NOT(GetProperties().Has(PENALTY_FACTOR))
+        << "No penalty factor (PENALTY_FACTOR) defined in property of SbmDirichletLaplacianCondition" << std::endl;
+    return 0;
+}
+
+void SbmDirichletLaplacianCondition::EquationIdVector(
+    EquationIdVectorType& rResult,
+    const ProcessInfo& rCurrentProcessInfo
+) const

```

Comment:
```suggestion
    const ProcessInfo& rCurrentProcessInfo) const
```
(just to be consistent with the rest of the file)